src/graphqlGenerator.test.ts

```diff
@@ -580,6 +580,100 @@ describe('generateGraphQL', () => {
     `)
   })
 
+  it('should compile deeply nested unions', async () => {
+    const graphql = await compileSchema({
+      $schema: 'http://json-schema.org/draft-07/schema#',
+      title: 'Parking lot',
+      additionalProperties: false,
+      properties: {
+        vehicles: {
+          type: 'array',
+          items: { $ref: '#/$defs/vehicle' },
+        },
+      },
+      $defs: {
+        vehicle: {
+          type: 'object',
+          title: 'Vehicle',
+          oneOf: [{ $ref: '#/$defs/car' }, { $ref: '#/$defs/bike' }],
+        },
+        car: {
+          type: 'object',
+          oneOf: [{ $ref: '#/$defs/passangerCar' }, { $ref: '#/$defs/lorry' }],
```

Comment:
a typo here: "pass _a_ nger" -> "pass _e_ nger" 🔍

Reply:
You really have hawk eyes when it comes to code reviews 🔍  I have missed that 🥲